File: Project Euler/Largest Palindrome Product/pedro_chagas_5485.py
def reverse_string(produto) -> str:
    return str(produto)[::-1]


def check_palindrom(produto) -> bool:
    return str(produto) == reverse_string(produto)


def update_largest_palindrom(produto, largest) -> int:
    if not check_palindrom(produto) or produto <= largest:
        return largest
    return produto


def solve() -> None:
    largest_palindrom: int = 0
    for first_product in range(100, 1000):
        for second_product in range(100, 1000):
            produto: int = first_product * second_product
            largest_palindrom = update_largest_palindrom(
                produto, largest_palindrom
            )

    print(largest_palindrom)


if __name__ == '__main__':
    solve()

File: Project Euler/Largest Palindrome Product/pedro_chagas_5485.rs
fn reverse_string(produto : &str) -> String {
    return produto.chars().rev().collect::<String>();
}

fn check_palindrom(produto : &str) -> bool {
    return produto == reverse_string(produto);
}

fn helper(p : u64, largest : u64) -> u64 {
    let pString : String = (p).to_string();
    if !check_palindrom(&pString) {
        return largest;
    }
    if p >= largest {
        return p
    }
    return largest;

}

fn solve() -> u64{
    let mut largest : u64 = 0;
    for i in 100 .. 1000 {
        for k in 100 .. 1000 {
            let p : u64 = i*k;
            largest = helper(p, largest);
        }
    }
    return largest;
}

fn main() {
    println!("{}",solve());

}
